eodag/api/product/_product.py

```diff
@@ -360,17 +350,10 @@ def download(
             else self.downloader_auth
         )
 
-        self.remote_location = urllib.parse.unquote(self.remote_location)
-        try:
-            # resolve remote location if needed with downloader configuration
-            self.remote_location = self.remote_location % vars(self.downloader.config)
-        except ValueError as e:
-            logger.debug(
-                f"Could not resolve product.remote_location ({self.remote_location})"
-                f" in download: {str(e)}"
-            )
-        if not self.location.startswith("file"):
-            self.location = urllib.parse.unquote(self.location)
+        # resolve remote location if needed with downloader configuration
+        self.remote_location = self.remote_location.replace("%", "%%") % vars(
+            self.downloader.config
+        )
```

Comment:
this won't work if remote location needs to be resolved.

But we may remove this block as this is already done in `register_download`